lib/registry/registry.rb

```diff
@@ -49,16 +51,15 @@ def paginate_doget(url)
         response = doget(url)
         yield response
 
-        link_header = response.headers[:link]
-        break unless link_header
-
-        url = parse_link_header(link_header)[:next]
+        link_header = response.headers[:link] or break
+        next_url = parse_link_header(link_header)[:next] or break
+        url = URI.join(response.request.url, next_url) # Interpret the next link relative to the current URL.
```

Comment:
This makes sense, based on your description. It would be worth adding a larger comment here, something like, "the `next` url from the link header could be relative to the current URL, or absolute. URI.join handles both cases cleanly." Or something like that.

Reply:
Here you go. I have made sure in the unit tests that both cases are checked.